app/Console/Command/UpdatePlaceLocality.php

```diff
@@ -76,33 +98,42 @@ public function configure(): void
 
     protected function execute(InputInterface $input, OutputInterface $output): ?int
     {
+        $itemType = new ItemType($input->getArgument(self::ITEM_TYPE));
+        if ($itemType->sameAs($itemType::event())) {
+            $output->writeln('<error>Only itemType place or organizer are accepted.</error>');
+            return self::FAILURE;
+        }
         $query = $input->getArgument(self::QUERY);
         $newLocality = $input->getArgument(self::NEW_LOCALITY);
 
         if ($query === null || $newLocality === null) {
-            $output->writeln('<error>Missing argument, the correct syntax is: place:update-locality "sapi3_query" "new_locality"</error>');
+            $output->writeln('<error>Missing argument, the correct syntax is: place:update-locality "item_type" "sapi3_query" "new_locality"</error>');
             return self::FAILURE;
         }
 
         $query = str_replace('q=', '', $query);
 
-        $count = $this->searchResultsGenerator->count($query);
+        $count = $this->searchPlaceResultsGenerator->count($query);
 
         if ($count <= 0) {
-            $output->writeln('<error>No places found</error>');
+            $output->writeln(sprintf('<error>No %ss found</error>', $itemType->toString()));
             return self::SUCCESS;
         }
 
-
         if (!$this->askConfirmation($input, $output, $count)) {
             return self::SUCCESS;
         }
 
-        foreach ($this->searchResultsGenerator->search($query) as $place) {
+        $searchResultsGenerator = $itemType->sameAs(ItemType::place()) ?
```

Comment:
Just make a helper method to get the correct `searchResultsGenerator`. You can do the same for `documentRepository`